### PR TITLE
[Orc-Binary-Cache-03] Enable ORC binary cache

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

After #1146 , Finally, enable ORC binary cache

## How was this patch tested?

unit tests, integration tests, manual tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zhixingheyi-tian** [comment]
@ivoson @LuciferYang 

Please have a review

**Comment by ivoson** [comment]
> @ivoson @LuciferYang
> 
> ```
>        Please have a review.
> ```
> 
> Thanks.

sure, will do soon

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/FilterSuite.scala [hunk 1]
```diff
@@ -293,6 +293,21 @@ class FilterSuite extends QueryTest with SharedOapContext with BeforeAndAfterEac
...
+    sql("insert overwrite table orc_test select * from t")
+    withIndex(TestIndex("orc_test", "index1")) {
+      sql("create oindex index1 on orc_test (a)")
+      withSQLConf(OapConf.OAP_ORC_BINARY_DATA_CACHE_ENABLED.key -> "true") {
+        sql("SELECT * FROM orc_test WHERE b = '1'")
+
+        checkAnswer(sql("SELECT * FROM orc_test WHERE  b = 'this is test 2'"),
+          Row(2, "this is test 2") :: Nil)
+      }
+    }
```
Comment (ivoson) [inline src/test/scala/org/apache/spark/sql/execution/datasources/oap/FilterSuite.scala]: can we add some check for the cache data here to make sure binary cache works?

File: src/main/scala/org/apache/spark/sql/internal/oap/OapConf.scala [hunk 2]
```diff
@@ -75,7 +75,7 @@ object OapConf {
       .doc("PartedByValueStatistics gives statistics with the value interval, default 5")
       .intConf
       .createWithDefault(5)
-
+  OAP_ORC_DATA_CACHE_ENABLED
```
[the remark this answers is not in the document]
Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/internal/oap/OapConf.scala]: Done

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/FilterSuite.scala (hunk 1 above)
Comment (zhixingheyi-tian) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/FilterSuite.scala]: Added

**Pull request merged**